python/graphstorm/gconstruct/utils.py

```diff
@@ -30,6 +30,112 @@
 from ..utils import sys_tracker
 from .file_io import HDF5Array
 
+SHARED_MEM_OBJECT_THRESHOLD = 1.9 * 1024 * 1024 * 1024 # must < 2GB
+SHARED_MEMORY_CROSS_PROCESS_STORAGE = "shared_memory"
+PICKLE_CROSS_PROCESS_STORAGE = "pickle"
+
+def _to_shared_memory(data):
+    """ Move all tensor objects into torch shared memory
+
+    Parameters
+    ----------
+    data: dict/tuple/list of tensors
+        Data returned by user_parser
+    """
+    if th.is_tensor(data):
+        return data.share_memory_()
+    elif isinstance(data, np.ndarray):
+        # only handle data that can be converted to torch tensor
+        if data.dtype in [np.float64, np.float32, np.float16,
+                          np.complex64, np.complex128, np.int64,
+                          np.int32, np.int16, np.int8,
+                          np.uint8, np.bool]:
+            return th.tensor(data).share_memory_()
+        return data
```

Comment:
what does this mean? the numpy array is returned directly? does it mean we give up in this case?

Reply:
Yes. In this case, we do nothing.
This happens, when data is a numpy.object_.

Reply:
should we report an error for this case?

Reply:
In this case, we just fallback to use pickle to pack the object.
I believe the most memory consuming objects are np.arrays which are handed in L53